fuel/transformers/__init__.py

```diff
@@ -619,3 +619,29 @@ def get_data(self, request=None):
             raise ValueError
         data = next(self.child_epoch_iterator)
         return data
+
+
+class SourcesSubsetTransformer(Transformer):
+    """Selects a subset of the stream sources.
+
+    Parameters
+    ----------
+    data_stream : :class:`DataStream` or :class:`Transformer`.
+        The data stream.
+    sources : tuple of strings
+        The names of the data sources returned by this transformer.
+        Must be a subset of the sources given by the stream.
+
+    """
+    def __init__(self, data_stream, sources):
+        super(SourcesSubsetTransformer, self).__init__(data_stream)
+        self.sources = sources
+        self._source_ids = [self.data_stream.sources.index(source)
```

Comment:
I suggest that before doing this we check that all sources are available in `data_stream` and raise an informative error message otherwise.